### PR TITLE
Feature/reports

### PR DESCRIPTION
**Pull request by abubakarazeem**

### CODE REVIEW

File: app/controllers/reports_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class ReportsController < ApplicationController
+  authorize_resource class: false
+
+  def issues
+    @issues = Issue.includes(:project, :assignee, :creator, :issue_state, :issue_type).all
```
Comment (Anmol79) [inline app/controllers/reports_controller.rb]: use current_tenant

File: app/views/reports/_issue_history_nested_table_for_attribute.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<table class="table table-sm">
+  <% audit.audited_changes.keys.each do |key| %>
+    <tr>
+      <td>
+        <% if assignee_id_key?(key) %>
```
Comment (Anmol79) [inline app/views/reports/_issue_history_nested_table_for_attribute.html.erb]: create a helper to return labels

File: app/helpers/reports_helper.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module ReportsHelper
+  def issue_priority(issue)
+    Issue::PRIORITY.key(issue.priority)
+  end
+
+  def assignee_id_key?(key)
```
Comment (Anmol79) [inline app/helpers/reports_helper.rb]: remove

File: app/helpers/reports_helper.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    key == 'assignee_id'
+  end
+
+  def display_user_name(id)
+    User.find(id).name
```
Comment (Anmol79) [inline app/helpers/reports_helper.rb]: current_tanant.users.find

File: app/models/user.rb [hunk 5]
```diff
@@ -89,12 +89,12 @@ def name
...
-  def name
-    "#{first_name} #{last_name}"
+  def admin?
+    role.name == Role::ROLES[:administrator]
   end
 
-  def admin?
-    role_id == Role.admin.id
+  def full_name
+    first_name + ' ' + last_name
```
Comment (Anmol79) [inline app/models/user.rb]: string interpolation

File: app/views/reports/_issue_audit_row.html.erb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<% if audit.action == 'update' %>
+  <tr>
+    <td><%= audit_counter%></td>
```
Comment (Anmol79) [inline app/views/reports/_issue_audit_row.html.erb]: space

File: app/views/reports/_issues_report.html.erb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<tr>
+  <td><%= link_to(issue.title, '#') %></td>
+  <td><%= link_to(issue.project.title, project_path(issue.project)) %></td>
+  <td><%= link_to(issue.assignee.full_name, member_path(issue.assignee)) %></td>
+  <td><%= link_to(issue.creator.full_name, member_path(issue.creator)) %></td>
+  <td><%= issue.issue_state.name %></td>
+  <td><%= issue.issue_type.name %></td>
+  <% if issue.priority == 0 %>
```
Comment (Anmol79) [inline app/views/reports/_issues_report.html.erb]: helper to return the badges based on priority and constant instead 0,1

File: config/routes.rb [hunk 8]
```diff
@@ -41,6 +41,9 @@
     confirmations: 'users/confirmations'
   }
 
+  get 'reports/issues', to: 'reports#issues', as: 'issues_report'
+  get 'issues/:issue_id/reports/issue_history', to: 'reports#issue_history', as: 'issue_history_report'
```
Comment (Anmol79) [inline config/routes.rb]: issue history report should be of all issues with a filter on issue name

**Comment by abubakarazeem** [comment]
review: cycle 0

**Pull request merged**